client/src/components/Voting.tsx

```diff
@@ -0,0 +1,290 @@
+    (candidate, index) => {return {
+      id: candidate.id,
+      name: candidate.name,
+      presentation: candidate.presentation,
+      symbolic: candidate.symbolic
+    }}
+  ).sort((a: candidateInfo, b: candidateInfo) => {
+    const aKey = keys.get(a.id) ?? [1, 1]
+    const bKey = keys.get(b.id) ?? [1, 1]
+    return compareList(aKey, bKey)
+  }))
+
+  const { classes, cx } = useStyle()
+  const [displayIndex, setDisplayIndex] = useState<Map<string, string>>(new Map<string, string>())
+  const [disabled, setDisabled] = useState(true)
+  const [hash, setHash] = useState<string | null>(null)
+  const [error, setError] = useState<string | null>(null)
+  const { authHeader } = useAuthorization()
+  const form = useForm({
+    initialValues: {
+      secret: "",
+      ranking: voteOrder.map((c) => c.id)
+    },
+    validate: {
+      secret: (s: string) => (s.length < 10 ? "Secret should be at least 10 characters long" : null)
+    }
+  })
+
+  const updateDisplayIndex = useCallback((indexes: Map<string, number>) => {
+    let firstSymbolic = voteOrder.length
+    voteOrder.forEach((candidate) => {
+      if (candidate.symbolic) firstSymbolic = Math.min(firstSymbolic, indexes.get(candidate.id) ?? firstSymbolic)
+    })
+    setDisplayIndex(new Map(voteOrder.map((candidate) => {
+      let index = indexes.get(candidate.id) ?? 0
+      return [candidate.id, `${firstSymbolic < index ? "-" : index + 1}`]
+    })))
+  }, [setDisplayIndex, voteOrder])
+
+  const handleItemUpdate = useCallback(({ destination, source }: DropResult) => {
+    if (destination == undefined) return;
+    const adjustedIndexes = new Map(voteOrder.map((candidate, index) => {
+      if (source.index == destination.index) return [candidate.id, index]
+      if (index == source.index) return [candidate.id, destination.index]
+      return [candidate.id, (source.index < destination.index ?
+        index - (index > source.index ? 1 : 0) + (index > destination.index ? 1 : 0) :
+        index + (index >= destination.index ? 1 : 0) - (index >= source.index ? 1 : 0)
+      )]
+    }))
+    updateDisplayIndex(adjustedIndexes)
+  }, [voteOrder])
+  
+  const handleItemDrop = useCallback(({ destination, source }: DropResult) => {
+    voteOrderHandlers.reorder({
+      from: source.index,
+      to: destination ? destination.index : source.index
+    })
+  }, [voteOrderHandlers])
+
+  const submitForm = useCallback(form.onSubmit((values) => {
+    axios.post(`/api/election/${props.election.id}/vote`, values, {
+      headers: authHeader
+    }).then((res) => {
+      setHash(res.data)
+      setDisabled(true)
+    }).catch((reason) => {
+      setError(reason.message)
+    })
+  }), [form])
+
+
+  useEffect(() => {
+    axios(`/api/election/${props.election.id}/has-voted`, {
+      headers: authHeader
+    }).then((res) => {
+      setDisabled(res.data == "true")
+    }).catch((reason) => {
```

Comment:
Reason is unused.
Consider copying the GA, prettierrc, eslintrc from betting hd.

Reply:
Or maybe not the eslint config. It is pretty bad I think